main.go

```diff
@@ -190,43 +212,56 @@ func main() {
 	}
 	filePath := os.Args[1]
 	debug("## Loading " + filePath)
+	
+	var test Test
+	var summary Summary
+	var subsetPartition map[int][]int
+
+	readTestFile(filePath, &test)
+	validate(&test, &subsetPartition)
```

Comment:
I agree with moving partition out of validate.  I can see how validate returning a boolean could be more standard and clear in terms of reading the code, however I think it is important for validate to return a specific error message which lets the test writer know what they have done wrong to making fixing the test easier, so I am inclined to return an error from validate.